feat: add information on out-of-battle effects to move & ability info

mudkipscience opened the pull request.

Favna reviewed (changes requested):
In abilities remove all the "from Pokemon emerald onward", "in geneartion X", etc. *Except* when it's exclusive to a generation(imem "only in Pokemon emerald" or "In Pokemon Sword & Shield").

Not even Bulbapedia is consistent and the inconsistency in this code is really annoying me. Take Snow Cloak for example and how it could've been listed as "From Generation IV onwards, ".

Favna commented on src/assets/moves.ts:
```diff
@@ -7855,7 +7867,8 @@ export default new GraphQLCollection<string, Pokemon.Move>([
       type: 'Normal',
-      contestType: 'Cute'
+      contestType: 'Cute',
+      isFieldMove: "Tthe Pokémon cuts some of its HP to recover another Pokémon's HP. It can be learned by Miltank, Skiddo and Gogoat by leveling up."
```
Tthe -> The

Favna commented on src/assets/moves.ts:
```diff
@@ -4109,7 +4118,8 @@ export default new GraphQLCollection<string, Pokemon.Move>([
       contestType: 'Clever',
-      maxMovePower: 130
+      maxMovePower: 130,
+      isFieldMove: 'The Pokemon can fly the player to any previously visited town/landmark. Fly can be taught to Pokemon with HM02 in all generations.'
```
"Pokemon" -> "Pokémon"

Also check if the same typo is elsewhere in items & abilities please. I'm pretty sure I did the rest a long while ago.

mudkipscience commented:
All done. Fixed some spelling errors, added the accent to Pokémon and removed the useless "from Pokémon Emerald onwards" text. I kept the "In Sword and Shield" stuff though, since currently those field effects only apply to those games.

mudkipscience commented:
tests should be good now
